File: 2018/pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        string k = argv[1];
        int keylen = strlen(k);
        for (int i = 0; i < keylen; i++)
        {
            if (!isalpha(k[i]))
            {
                printf("has non-alphabetical character\n");
                return 1;
            }
            else
            {
                k[i] = toupper(k[i]);
            }
        }

        string ptext = get_string("plaintext:  ");
        printf("ciphertext: ");
        for (int i = 0, j = 0, n = strlen(ptext); i < n; i++)
        {
            if (isalpha(ptext[i]))
            {
                int m = k[j] - 'A';
                if (isupper(ptext[i]))
                {
                    printf("%c", (ptext[i] + m > 90 ? ptext[i] + m - 26 : ptext[i] + m));
                }
                else
                {
                    printf("%c", (ptext[i] + m > 122 ? ptext[i] + m - 26 : ptext[i] + m));
                }
                j++;
                if (j == keylen)
                {
                    j = 0;
                }
            }
            else
            {
                printf("%c", ptext[i]);
            }
        }
        printf("\n");
    }
    else
    {
        printf("Usage: ./vigenere k\n");
        return 1;
    }
}

File: 2018/pset6/vigenere/vigenere.py
from cs50 import get_string
import sys

if len(sys.argv) > 2:
    sys.exit(1)

key = sys.argv[1]
if key.isalpha():
    key = key.upper()
else:
    print("has non-alphabetical character")
    sys.exit(1)

pt = get_string("plaintext:  ")
print("ciphertext: ", end='')

#print(key)
i = 0
for c in pt:
    if c.isalpha():
        m = ord(key[i]) - ord('A')
        if c.isupper():
            print(chr(ord(c) + m - 26) if ord(c) + m > 90 else chr(ord(c) + m), end='')
        else:
            print(chr(ord(c) + m - 26) if ord(c) + m > 122 else chr(ord(c) + m), end='')
        i += 1
        if i == len(key):
            i = 0
    else:
        print(c, end='')
print("")

